src/dyndns.py

```diff
@@ -170,6 +185,65 @@ def dyndns_subscribe(operation_logger, domain=None, key=None):
     logger.success(m18n.n("dyndns_registered"))
 
 
+@is_unit_operation()
+def dyndns_unsubscribe(operation_logger, domain, password=None):
+    """
+    Unsubscribe from a DynDNS service
+
+    Keyword argument:
+        domain -- Full domain to unsubscribe with
+        password -- Password that is used to delete the domain ( defined when subscribing )
+    """
+
+    operation_logger.start()
+    
+    from yunohost.utils.password import assert_password_is_strong_enough
+
+    # Ensure sufficiently complex password
+    if Moulinette.interface.type == "cli" and not password:
+        password = Moulinette.prompt(
+        m18n.n("ask_password"), is_password=True, confirm=True
+    )
+    assert_password_is_strong_enough("admin", password)
+
+    # '165' is the convention identifier for hmac-sha512 algorithm
+    # '1234' is idk? doesnt matter, but the old format contained a number here...
+    key_file = f"/etc/yunohost/dyndns/K{domain}.+165+1234.key"
+
+    import requests  # lazy loading this module for performance reasons
+
+    # Send delete request 
+    try:
+        r = requests.delete(
+            f"https://{DYNDNS_PROVIDER}/domains/{domain}",
+            data={"recovery_password":hashlib.sha256((str(domain)+":"+str(password).strip()).encode('utf-8')).hexdigest()},
+            timeout=30,
+        )
+    except Exception as e:
+        raise YunohostError("dyndns_unregistration_failed", error=str(e))
+
+    if r.status_code == 200: # Deletion was successful
+        rm(key_file, force=True)
+        # Yunohost regen conf will add the dyndns cron job if a key exists
+        # in /etc/yunohost/dyndns
+        regen_conf(["yunohost"])
+
+        # Add some dyndns update in 2 and 4 minutes from now such that user should
+        # not have to wait 10ish minutes for the conf to propagate
+        cmd = (
```

Comment:
We don't need that call to dns push

Reply:
Fixed in https://github.com/YunoHost/yunohost/pull/1472/commits/dd51adcd3f467968c2663f985832e43395d0b995